### PR TITLE
T4F-76: loan module

### PR DESCRIPTION
**Pull request by ChristianPerez34**
Created reducer with necessary actions to add and view loans. There are a few working modals and others that do nothing.

### CODE REVIEW

File: src/views/Loans/ManageLoans/ViewLoan.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                                    </p>
+                                </GridItem>
+                            </GridContainer>
+                            <Clearfix/>
+                        </CardBody>
+                    </Card>
+                </GridItem>
+            </GridContainer>
+            <AmortizationTable data={dummyAmortizationSchedule}/>
+            <NotesTable data={loanNotes}/>
```
Comment (cfboy) [inline src/views/Loans/ManageLoans/ViewLoan.js:235]: Las tablas de Notes y Attachments se pueden poner en una misma fila. Intenta ponerlas dentro de un GridItem de md={6} para que no cojan tanto espacio.

File: src/components/Modal/AddNoteModal.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                    >
+                        <Close className={classes.modalClose}/>
+                    </Button>
+                    <h4 className={classes.modalTitle}>Add Note</h4>
+                </DialogTitle>
+                <DialogContent
+                    id="modal-slide-description"
+                    className={classes.modalBody}
+                >
+                    <CustomInput
```
Comment (cfboy) [inline src/components/Modal/AddNoteModal.js:91]: Para añadir notas creo que lo mejor es poner un TextArea, pero por ahora se puede dejar así.

File: src/components/Modal/AddOverchargesModal.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                    >
+                        <Close className={classes.modalClose}/>
+                    </Button>
+                    <h4 className={classes.modalTitle}>Add Charge</h4>
+                </DialogTitle>
+                <DialogContent
+                    id="modal-slide-description"
+                    className={classes.modalBody}
+                >
+                    <CustomInput
```
Comment (cfboy) [inline src/components/Modal/AddOverchargesModal.js:96]: Para el add charges y el add payments no hacen falta mas campos? Para especificar fecha o tipo ?

File: src/components/Modal/AddAttachmentModal.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                    >
+                        <Close className={classes.modalClose}/>
+                    </Button>
+                    <h4 className={classes.modalTitle}>Add Attachment</h4>
+                </DialogTitle>
+                <DialogContent
+                    id="modal-slide-description"
+                    className={classes.modalBody}
+                >
+                    <Dropzone
```
Comment (cfboy) [inline src/components/Modal/AddAttachmentModal.js:92]: Aquí no hace falta un campo para especificar el tipo de attachment? Por ejemplo se puede poner un dropdown con algo asi: Fotos de Propiedad, documentos oficiales, Otros

File: src/components/Modal/AddNoteModal.js (hunk 2 above)
Comment (ChristianPerez34) [reply src/components/Modal/AddNoteModal.js:91]: En Material-UI para hacer multilines vi que lo que hacen es setear el prop de multiline a true

File: src/components/Modal/AddOverchargesModal.js (hunk 3 above)
Comment (ChristianPerez34) [reply src/components/Modal/AddOverchargesModal.js:96]: Los unicos modals que implemente fueron los de Notes y Attachments. Eso fue lo que les comente en la reunion. Los demas estan incompletos para completarse en la fase de integracion. 
Añado lo que me dijistes en un todo para asi no perder el comment

File: src/components/Modal/AddAttachmentModal.js (hunk 4 above)
Comment (ChristianPerez34) [reply src/components/Modal/AddAttachmentModal.js:92]: Vale me suena bien. Yo pensaba que tipo de attachment se referia si era .jpg, .pdf etc.

**Comment by alejoreyes96** [comment]
image

Not a bug, but maybe these shortcuts should be added at the top of the respective table (for notes and attachments)

Also, the 'x' button on Loan Information card, will it be used?

**Comment by alejoreyes96** [comment]
I think you should add an arrow to navigate back to the previous screen once you are inside Loan Summary, like in Customers module.

image

**Comment by alejoreyes96** [comment]
image
Same as all other modules, this Header is a button, but when the users click it, nothing happens. It is misleading. Should either add a redirect for Loans or have it become disabled or static

**Comment by alejoreyes96** [comment]
> image
> Same as all other modules, this Header is a button, but when the users click it, nothing happens. It is misleading. Should either add a redirect for Loans or have it become disabled or static

Also, perhaps you should include the status field of the loans in between hazar Exp Date and Actions, to know which the user can or can't delete. 

Another thing is that delete buttons should be fully implemented for HH, whenever required.

**Comment by ChristianPerez34** [comment]
> image
> 
> Not a bug, but maybe these shortcuts should be added at the top of the respective table (for notes and attachments)
> 
> Also, the 'x' button on Loan Information card, will it be used?

Deletes have not been implemented in favor of leaving it for the implementation phase, but if you think it should be there then I'll add it.

The buttons were moved there based on @cfboy recomendation when restructuring UI for loans view

**Comment by ChristianPerez34** [comment]
> image
> Same as all other modules, this Header is a button, but when the users click it, nothing happens. It is misleading. Should either add a redirect for Loans or have it become disabled or static

I think this is automatically generated from the dashboard

**Comment by ChristianPerez34** [comment]
> I think you should add an arrow to navigate back to the previous screen once you are inside Loan Summary, like in Customers module.
> 
> image

Ok

**Comment by ChristianPerez34** [comment]
> > image
> > Same as all other modules, this Header is a button, but when the users click it, nothing happens. It is misleading. Should either add a redirect for Loans or have it become disabled or static
> 
> Also, perhaps you should include the status field of the loans in between hazar Exp Date and Actions, to know which the user can or can't delete.
> 
> Another thing is that delete buttons should be fully implemented for HH, whenever required.

Sounds good to me

**Comment by alejoreyes96** [comment]
> > image
> > Not a bug, but maybe these shortcuts should be added at the top of the respective table (for notes and attachments)
> > Also, the 'x' button on Loan Information card, will it be used?
> 
> Deletes have not been implemented in favor of leaving it for the implementation phase, but if you think it should be there then I'll add it.
> 
> The buttons were moved there based on @cfboy recomendation when restructuring UI for loans view
Got it. 

The deletes i mention it because it is likely that it will draw attention on the Happy Hour. Besides that, I agree that we can leave it for the next phase to implement it correctly

**Comment by alejoreyes96** [comment]
> > image
> > Same as all other modules, this Header is a button, but when the users click it, nothing happens. It is misleading. Should either add a redirect for Loans or have it become disabled or static
> 
> I think this is automatically generated from the dashboard

Ohh got it

**Pull request merged**